feat: Improve self-monitor troubleshooting guides (#1142)
kyma-project/telemetry-manager: docs/user/04-metrics.md

### No Metrics Arrive at the Backend

> Symptom: Incorrect backend endpoint configuration (e.g., using the wrong authentication credentials) or the backend being unreachable.

Reviewer: ```suggestion
Symptom: No metrics arrive at the backend.

Cause: Incorrect backend endpoint configuration (e.g., using the wrong authentication credentials) or the backend being unreachable.
```

Remedy: 
- Check the `telemetry-metric-gateway` Pods for error logs by calling `kubectl logs -n kyma-system {POD_NAME}`.
- Check if the backend is up and reachable.

### Not All Metrics Arrive at the Backend

Symptom: The backend is reachable and the connection is properly configured, but some metrics are refused.

> Cause: It can happen due to a variety of reasons. For example, a possible reason may be that the backend is limiting the ingestion rate.

Reviewer: ```suggestion
Cause: It can happen due to a variety of reasons, you need to check the logs. For example, a possible reason may be that the backend is limiting the ingestion rate.
```

Remedy:

> - Check the `telemetry-metric-gateway` Pods for error logs by calling `kubectl logs -n kyma-system {POD_NAME}`. If backend is limiting the rate by refusing metrics, try the options desribed in [Gateway Buffer Filling Up](#gateway-buffer-filling-up)

Reviewer: ```suggestion
1. Check the `telemetry-metric-gateway` Pods for error logs by calling `kubectl logs -n kyma-system {POD_NAME}`. 
2. If backend is limiting the rate by refusing metrics, try the options desribed in [Gateway Buffer Filling Up](#gateway-buffer-filling-up).
3. Otherwise, take the actions appropriate to the cause indicated in the logs.
```

Do we really want that they only check the Pod logs? Not the backend logs?

<!-- tabs:end -->

### Gateway Buffer Filling Up

Cause: The backend export rate is too low compared to the gateway ingestion rate.

Remedy:

- Option 1: Increase ingestion rate capabilities in your backend. For example, by scaling out the SAP Cloud Logging instances.

> - Option 2: Decrease the amount of metrics sent to the gateway (e.g., by disabling certain inputs or applying namespace filters in your MetricPipeline).

Reviewer: ```suggestion
- Option 2: Decrease the amount of metrics sent to the gateway (for example, by disabling certain inputs or applying namespace filters in your MetricPipeline).
```
